File: backend/src/main/java/com/advisor/entity/Notification.java
package com.advisor.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "notifications")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    @Id
    private String id;
    
    // The user who receives the notification
    private String recipientId;
    
    // The user who triggered the notification
    private String senderId;
    private String senderName;
    private String senderAvatarUrl;
    
    // Type of notification: LIKE, COMMENT, FOLLOW_REQUEST, FOLLOW_ACCEPT, SHARE, APPLICATION
    private String type;
    
    // The actual text payload ("John Doe liked your post")
    private String message;
    
    // ID to deep link to (postId, etc)
    private String relatedEntityId;
    
    private boolean isRead = false;
    
    @CreatedDate
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: career_advisor_flutter/lib/models/notification.dart
class AppNotification {
  final String id;
  final String recipientId;
  final String senderId;
  final String senderName;
  final String? senderAvatarUrl;
  final String type;
  final String message;
  final String? relatedEntityId;
  final bool isRead;
  final DateTime createdAt;

  AppNotification({
    required this.id,
    required this.recipientId,
    required this.senderId,
    required this.senderName,
    this.senderAvatarUrl,
    required this.type,
    required this.message,
    this.relatedEntityId,
    required this.isRead,
    required this.createdAt,
  });

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    return AppNotification(
      id: json['id'] ?? '',
      recipientId: json['recipientId'] ?? '',
      senderId: json['senderId'] ?? '',
      senderName: json['senderName'] ?? 'Someone',
      senderAvatarUrl: json['senderAvatarUrl'],
      type: json['type'] ?? 'SYS',
      message: json['message'] ?? '',
      relatedEntityId: json['relatedEntityId'],
      isRead: json['read'] ?? false,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }
}
